src/components/CommentsSection.jsx

```diff
@@ -1,26 +1,68 @@
-import { useState, useEffect } from "react";
+import { useState, useEffect, useContext } from "react";
 import Comment from "./Comment";
 import Loading from "./Loading";
-import { fetchComments } from "../../utils/api";
+import { fetchComments, postComment } from "../../utils/api";
+import { UserContext } from "../context/User";
 
 export default function CommentsSection({ articleId }) {
+  const { user, setUser } = useContext(UserContext);
   const [comments, setComments] = useState({});
   const [loading, setLoading] = useState(true);
+  const [newComment, setNewComment] = useState("");
+  const [submitted, setSubmitted] = useState(false);
+  const [error, setError] = useState(null);
 
   useEffect(() => {
     fetchComments(articleId).then((commentsData) => {
       setComments(commentsData);
       setLoading(false);
     });
-  }, []);
+  }, [submitted]);
+
+  const handleChange = ({ target: { value } }) => {
+    setNewComment(value);
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    setError(null);
+    postComment(user, newComment, articleId)
+      .then(() => {
+        setNewComment("");
+        setSubmitted(true);
+      })
+      .catch((err) => {
```

Comment:
does this error handling work? Have you seen it work? I tried forcing an error and the app crashed